### ISSUE TITLE
Many Bugs

### ISSUE THREAD

**Issue by mabbamOG** [torhve/weechat-matrix-protocol-script#76]
there are many bugs in this script.
the help command does not work
some join commands do not work and complain about api errors
everything is very laggy. matrix takes complete control over weechat and even heavily delays shutdown times
some joined channels do not appear to actually be working (such as over freenode bridge)

the plugin is not reliable at all :\

**Comment by hiqua** [comment]
The riot app is just as slow, my bet would be that it's rather due to the slow server than the plugin itself. Some of the API errors are related, i.e. the timeout is not lenient enough for the current performance of the main server.

**Comment by grahamperrin** [comment]
A quick comparison: [Quaternion](https://matrix.org/docs/projects/client/quaternion.html) spends a while, after launch, synchronising … after that waiting period, things are not slow at all.

**Comment by ara4n** [comment]
yeah, this is nothing to do with the server, and riot's reasons for slowness are entirely unrelated. instead the problem is that the script sometimes sits chewing 100% CPU for seconds on end, locking the whole app, and it's unclear why.

**Comment by ara4n** [comment]
this is basically a dup of https://github.com/torhve/weechat-matrix-protocol-script/issues/50

**Comment by ara4n** [comment]
...and #30
